crates/ironrdp-rdcleanpath/src/lib.rs

```diff
@@ -323,7 +355,18 @@ impl TryFrom<RDCleanPathPdu> for RDCleanPath {
                 server_addr,
             }
         } else {
-            Self::Err(pdu.error.ok_or(MissingRDCleanPathField("error"))?)
+            let error = pdu.error.ok_or(MissingRDCleanPathField("error"))?;
+            if error.error_code == NEGOTIATION_ERROR_CODE {
+                if let Some(x224_pdu) = pdu.x224_connection_pdu {
+                    Self::NegotiationErr {
+                        x224_connection_response: x224_pdu.as_bytes().to_vec(),
+                    }
+                } else {
+                    Self::GeneralErr(error)
+                }
+            } else {
+                Self::GeneralErr(error)
+            }
```

Comment:
nitpick: I think this more concise version is more readable

```suggestion
            match (error.error_code, pdu.x224_connection_pdu) {
                (NEGOTIATION_ERROR_CODE, Some(x224_pdu)) => Self::NegotiationErr {
                    x224_connection_response: x224_pdu.as_bytes().to_vec(),
                },
                _ => Self::GeneralErr(error),
            }
```